overlord/snapstate/check_snap.go

```diff
@@ -542,7 +542,7 @@ func checkAndCreateSystemUsernames(si *snap.Info) error {
 
 	// then create
 	// TODO: move user creation to a more appropriate place like "link-snap"
-	extrausers := !release.OnClassic
+	extrausers := !release.OnClassic || release.OnTouch
```

Comment:
I wonder if beside exporting OnTouch, we wouldn't want to have something that says if extrausers should be used or not in itself,  we have other places that make this decision:

https://github.com/snapcore/snapd/blob/master/overlord/devicestate/users.go#L140 ,  that code is likely not triggered anyway on Classic systems but might not be a good idea to leave it in an inconsistent state.

Also I suppose you don't want to change this, as snapd is not managing the system in the Touch case?:
https://github.com/snapcore/snapd/blob/master/daemon/api_users.go#L204

Reply:
I could see two things:

- Your solution
- either the concept of a read-only system is described similarly through all those `OnClassic` and whatnot variables (ie `OnReadOnly`) but then again, what paths are read-only depends on the system (Touch and Core) have different ideas.

Reply:
I suppose a separate explicit  flag eg `SupportsExtraUsers` and consistent use in the places where the manipulate user accounts could be sufficient, even without the need to add a separate OnTouch variant (but test for it and set the flag accordingly in release.go)

Reply:
Should this `SupportsExtraUsers` also reside in `release.go` or do you have a better, still reachable place in mind?

Reply:
The way I see it is that we're very much past the point where adding an additional abstraction of a variable is overkill, compared to a sustainable-enough 3-liner PR. If you are fine with this, all things considered, I will rebase on top of recent master and force-push again.